system-addon/lib/TopSitesFeed.jsm

```diff
@@ -65,9 +69,15 @@ this.TopSitesFeed = class TopSitesFeed {
       }
     }
 
-    // Now, get a screenshot for every item
+    // Now, get a tippy top icon or screenshot for every item
     for (let link of links) {
       if (!link) { continue; }
+
+      // Check for tippy top icon.
+      link = this._tippyTopProvider.processSite(link);
```

Comment:
So, this seems like it could fail or give wrong results if `TippyTopProvider` hasn't finished adding to `_sitesByDomain`. I would guess the correct fix is to make `processSite` `async`. But we can skip this for now.